### PR TITLE
Display and allow to kill queued queries

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/connector/system/KillQueryProcedure.java [hunk 1]
```diff
@@ -65,13 +71,36 @@ public void killQuery(String queryId, String message)
...
+            if (!ADMINISTRATIVELY_KILLED.toErrorCode().equals(requireNonNull(queryManager.getQueryInfo(query).getErrorCode()))) {
+                throw new PrestoException(NOT_SUPPORTED, "Target query is already finished: " + queryId);
+            }
+            return;
+        }
+        catch (NoSuchElementException e) {
+        }
+
+        try {
+            checkState(dispatchManager.isPresent(), "No dispatch manager is set. kill_query procedure should be executed on coordinator.");
```
Comment (kokosing) [inline presto-main/src/main/java/io/prestosql/connector/system/KillQueryProcedure.java]: @dain is it possible that there could be a race that query is no longer stored in `dispatchManager`, but we have found it in `queryManager` either?

Maybe I should check `dispatchManager` first. Would be any race there?

**Comment by kokosing** [comment]
@dain I rebased and added tests for that so now it is much easier to understand what I actually fixed.

>

File: presto-main/src/main/java/io/prestosql/connector/system/KillQueryProcedure.java [hunk 2]
```diff
@@ -43,11 +45,13 @@
     private static final MethodHandle KILL_QUERY = methodHandle(KillQueryProcedure.class, "killQuery", String.class, String.class);
 
     private final QueryManager queryManager;
```
Comment (dain) [inline presto-main/src/main/java/io/prestosql/connector/system/KillQueryProcedure.java]: QueryManager is no longer used and should be removed

File: presto-main/src/main/java/io/prestosql/connector/system/KillQueryProcedure.java [hunk 3]
```diff
@@ -43,11 +45,13 @@
...
+    private final Optional<DispatchManager> dispatchManager;
 
     @Inject
-    public KillQueryProcedure(QueryManager queryManager)
+    public KillQueryProcedure(QueryManager queryManager, Optional<DispatchManager> dispatchManager)
```
Comment (dain) [inline presto-main/src/main/java/io/prestosql/connector/system/KillQueryProcedure.java]: Only bind this in the coordinator and then you don't need the optional

Comment (kokosing) [reply presto-main/src/main/java/io/prestosql/connector/system/KillQueryProcedure.java]: Then I would need to break `SystemConnectorModule` into several modules. Not sure what is better. I would refrain from this refactactor in this PR. Let me fix this as a follow up

**Comment by kokosing** [comment]
@dain Addressed your comments. Thank you for review!

**Pull request merged**